tools/mcp/dotnet/AzureSDKDevToolsMCP/Tools/SpecPullRequestTool.cs

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel;
+using AzureSDKDevToolsMCP.Helpers;
+using AzureSDKDevToolsMCP.Services;
+using ModelContextProtocol.Server;
+using Octokit;
+
+namespace AzureSDKDevToolsMCP.Tools
+{
+    [Description("TypeSpec pull request tools")]
+    [McpServerToolType]
+    public class SpecPullRequestTool(IGitHubService service, IGitHelper helper)
+    {
+        private readonly IGitHubService gitHubService = service;
+        private readonly IGitHelper gitHelper = helper;
+        readonly string TEST_IGNORE_TAG = "[TEST-IGNORE]";
+
+
+        [McpServerTool, Description("Connect to GitHub using personal access token.")]
+        public async Task<string> GetGitHubUserDetails()
+        {
+            var user = await this.gitHubService.GetGitUserDetails();
+            return user != null
+                ? $"Connected to GitHub as {user.Login}"
+                : "Failed to connect to GitHub. Please make sure to login to GitHub using gh auth login to connect to GitHub.";
+        }
+
+        [McpServerTool, Description("Get Pull Request Status: Get TypeSpec pull request status for a given pull request number.")]
+        public async Task<string> GetPullRequestStatus(int pullRequestNumber, string typeSpecProjectPath, string repoOwner = "Azure")
+        {
+            try
+            {
+                Console.WriteLine($"Getting pull request details for {pullRequestNumber}...");
```

Comment:
Can you start injecting ILogger early on? That will make it easier for us to refactor when/if we consume this as part of a larger MCP server project.